### PR TITLE
[3.6.x] DDF-351 Extension Point for Security Logging

### PR DESCRIPTION
**Pull request by cantstoptheunk**
- Adds a security logging function that can be used downstream.
- If a downstream application does not fill out a function definition then the code will no-op and do nothing.

### CODE REVIEW

**Review by andrewkfiedler — CHANGES_REQUESTED** [review]
If this is centered around the inspector visual, shouldn't we be tying it into that visual rather than the result selector?

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/result-selector/result-selector.tsx [hunk 1]
```diff
@@ -31,7 +31,6 @@ import { Elevations } from '../theme/theme'
 import useTheme from '@material-ui/core/styles/useTheme'
 import SelectionRipple from '../golden-layout/selection-ripple'
 import { ResultType } from '../../js/model/Types'
-import Extensions from '../../extension-points'
```
Comment (cantstoptheunk) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/result-selector/result-selector.tsx]: should probably not delete

File: ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/inspector/inspector-lazy.tsx [hunk 2]
```diff
@@ -22,14 +26,49 @@ const LazyInspector = ({ selectionInterface }: Props) => {
   React.useEffect(() => {
     selectionInterface.setSelectedResults(backboneModels)
   })
+  const conductAudit = () => {
+    let newSelectedIds = new Set<string>()
+    for (let key in selectedResults) {
```
Comment (cantstoptheunk) [inline ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/inspector/inspector-lazy.tsx]: call it id instead of key

Comment (cantstoptheunk) [reply ui-frontend/packages/catalog-ui-search/src/main/webapp/component/visualization/inspector/inspector-lazy.tsx]: object.keys() to grab the list of keys

**Review by criscrasbell — CHANGES_REQUESTED** [review]
This extension point is too specific. It will only have one purpose. I think we should put the audit logging code entirely in DDF-UI.

**Comment by andrewkfiedler** [comment]
Going to go a different route on this

**Pull request closed without merging**